### PR TITLE
[IMP] website_sale, website: ecommerce product checkout redesign

### PR DESCRIPTION
**Pull request by mano-odoo**
The goal of this PR is to redesign the product page and the overall checkout
flow (figma on task).

It introduces:

--- Overall --- 

- New column layout with an offset above xl screen
- Spacing improvement across the pages

--- Product Page ---

- Split in two columns view for 100_pc and none
- Reviewed cta_wrapper (add to cart, buy now, qty) responsiveness and sizing
- Product variant (SCSS cleaning, pill variant redesign, alignment)
- Sharing into a dropdown
- Refreshed wishlist and optional product modal
- Specification table being full width
- Grid view images ratio

--- Checkout flow ---

- New navigation flow with buttons inside the cart_summary template (introducing
	`cta_buttons_navigation` template)
- Reviewed tables accross the pages removing stripe and improving spacing
- Redesigned stepper wizard with a reusable template in `/website` 

task-3083707

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/static/src/scss/website_sale.scss [hunk 1]
```diff
@@ -784,3 +794,26 @@ a.no-decoration {
...
+}
+
+.o_total_border {
+    border: $border-width solid $border-color;
+    @include media-breakpoint-down(lg) {
+        border-width: 0;
+    }
+}
+
+.all_shipping {
```
Comment (stefanorigano) [inline addons/website_sale/static/src/scss/website_sale.scss:815]: This will target **any** `<div/>` inside, is it intentional?

Comment (stefanorigano) [reply addons/website_sale/static/src/scss/website_sale.scss:791]: can you apply the `overflow-x-auto` class where needed instead?

File: addons/website_sale/views/templates.xml [hunk 2]
```diff
@@ -1723,22 +1721,10 @@
...
-                        <a role="button" href='#' class="btn btn-block btn-primary">
-                            <i class='fa fa-check'></i> Ship to this address
-                        </a>
-                    </span>
-                    <span class='btn-ship' t-att-style="'' if not selected else 'display:none;'">
-                        <a role="button" href='#' class="btn btn-block btn-secondary">
-                            Select this address
-                        </a>
-                    </span>
+                    <a href="#" class="js_edit_address btn btn-link p-0 mt-2 fw-bold {{ 'o_disabled' if readonly else '' }}" role="button" title="Edit this address" aria-label="Edit this address">Edit</a>
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1727]: wrong attribute: `class` should be `t-attf-class` ;)

File: addons/website_sale/views/templates.xml [hunk 3]
```diff
@@ -1723,22 +1721,10 @@
                 </t>
                 <input type='submit'/>
             </form>
-            <div t-attf-class="card #{selected and 'border border-primary' or 'js_change_shipping'}">
+            <div t-attf-class="card position-relative #{selected and 'bg-primary border border-primary' or 'js_change_shipping'} h-100">
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1724]: minor: move `h-100` class before `#{...}`, for readability

File: addons/website_sale/views/templates.xml [hunk 4]
```diff
@@ -1756,26 +1742,26 @@
...
                                 <t t-set="hide_promotions">True</t>
                                 <t t-set="redirect" t-valuef="/shop/address"/>
                             </t>
                         </div>
-                        <div class="col-12 col-xl order-xl-1 oe_cart">
+                        <div class="col-12 col-lg-7 order-lg-1 oe_cart">
                             <div>
                                 <t t-if="mode == ('new', 'billing')">
-                                    <h2 class="o_page_header mt8">
+                                    <h3 class="mb-3">
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1754]: General note:  avoid customizing inline elements margin unless it's strictly necessary.
Let's keep in mind that the the user can change the font-family... if I switch to _Lobster_, for example, I would probably adapt margins accordingly... but these classes will prevent any change ;)

I'll let you check which margin adaptations are necessary

File: addons/website_sale/views/website_sale_delivery_templates.xml [hunk 5]
```diff
@@ -18,7 +18,7 @@
     <template id="payment_delivery_methods">
         <input t-att-value="delivery.id" t-att-id="'delivery_%i' % delivery.id" t-att-delivery_type="delivery.delivery_type" type="radio" name="delivery_type" t-att-checked="order.carrier_id and order.carrier_id.id == delivery.id and 'checked' or False"/>
         <label class="label-optional" t-field="delivery.name"/>
-        <t t-set='badge_class' t-value="'float-end ' + 'badge text-bg-secondary'" />
+        <t t-set='badge_class' t-value="'float-end fw-bold'" />
```
Comment (stefanorigano) [inline addons/website_sale/views/website_sale_delivery_templates.xml:21]: To define a variable was needed by previous implementations (see https://github.com/odoo/odoo/pull/105848/files#diff-83a28c43e60e845a7eb01351dd94670014c0aedc760d28b1d5dac8f6a0610fbaL21) but it doesn't seems the case anymore.

I think that `badge_class` can be removed and its values added to the following `<span>` directly.

File: addons/portal/static/src/xml/portal_chatter.xml [hunk 6]
```diff
@@ -4,7 +4,6 @@
         <t t-set="count" t-value="widget.get('message_count')"/>
         <div class="o_message_counter">
             <t t-if="count">
-                <span class="fa fa-comments" />
```
Comment (stefanorigano) [inline addons/portal/static/src/xml/portal_chatter.xml:7]: Why this change is needed?

File: addons/website_sale/static/src/scss/product_configurator.scss [hunk 7]
```diff
@@ -65,11 +65,29 @@
         border-color: $input-border-color;
     }
 
+    i {
```
Comment (stefanorigano) [inline addons/website_sale/static/src/scss/product_configurator.scss:68]: Better to target classes when possible,  `.fa` in this case ;)

File: addons/website_sale/static/src/scss/product_configurator.scss [hunk 8]
```diff
@@ -65,11 +65,29 @@
...
+        position: relative;
+        z-index: 1;
+        font-size: $font-size-base * .875;
+    }
+
+    .btn:hover::before {
+        content: "";
+        position: absolute;
```
Comment (stefanorigano) [inline addons/website_sale/static/src/scss/product_configurator.scss:76]:
```suggestion
        @include o-position-absolute(50%, $left: 50%);
```
;)

File: addons/website_sale/static/src/scss/website_sale.scss [hunk 9]
```diff
@@ -206,13 +191,21 @@ $o-wsale-products-layout-grid-gutter-width: min($grid-gutter-width / 2, $o-wsale
         }
     }
     .o_pricelist_dropdown, .o_sortby_dropdown {
-        min-width: fit-content;
+        //truncates the dropdown pricelist > 8rem
```
Comment (stefanorigano) [inline addons/website_sale/static/src/scss/website_sale.scss:194]: I don't understand why is needed... watchout that once switched from "Benelux" to "EUR" my dropdown disappeared

File: addons/website_sale/views/templates.xml [hunk 10]
```diff
@@ -967,18 +973,25 @@
...
+                                        <div id="o_price_container_div" t-attf-class="w-100 me-2 {{ '' if combination_info['base_unit_price'] else '' }}">
+                                            <t t-call="website_sale.product_price"/>
+                                            <small t-if="combination_info['base_unit_price']"
+                                                   class="ms-1 me-2 text-muted o_base_unit_price_wrapper d-none" groups="website_sale.group_show_uom_price">
+                                                <t t-call='website_sale.base_unit_price'/>
+                                            </small>
+                                        </div>
+                                        <div id="add_to_cart_wrap" t-attf-class="flex-grow-1 {{'d-none' if combination_info['prevent_zero_price_sale'] else 'd-inline-flex'}}">
+                                            <a data-animation-selector=".o_wsale_product_images"
+                                                role="button" id="add_to_cart"
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:989]: Detail: either you move each attribute in a dedicated line or you keep them all in one ;)
Some people do combine short attributes (like `role="button"` and `href="#"` in this case), but I'd advise against.

File: addons/website_sale/views/templates.xml [hunk 11]
```diff
@@ -967,18 +973,25 @@
...
+                                                t-attf-class="js_check_product btn btn-primary  a-submit flex-grow-1 {{'mw-100' if not hasBuyNow and o_product_width == '66_pc' else 'mw-50' if not hasBuyNow else ''}}" href="#"
+                                            >
+                                                <span>Add to cart</span>
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:992]: Is the `<span>`tag necessary?

File: addons/website_sale/views/templates.xml [hunk 12]
```diff
@@ -1009,12 +1022,12 @@
...
                             </div>
-                            <div id="o_product_terms_and_share" class="d-flex justify-content-between flex-column flex-md-row align-items-md-end mb-3">
-                            </div>
+                            <!-- terms and conditions change position according to the image layout -->
+                            <t t-if="not o_product_width_fw" t-call="website_sale.product_custom_text"/>
                         </div>
                     </div>
                 </section>
-                <div itemprop="description" t-field="product.website_description" class="oe_structure oe_empty mt16" id="product_full_description"/>
+                <div id="product_full_description" itemprop="description" t-field="product.website_description" class="oe_structure oe_empty mt16"/>
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1030]: Avoid unnecessary diffs like reordering attributes

File: addons/website/static/src/scss/website.scss [hunk 13]
```diff
@@ -1378,6 +1378,36 @@ header {
     }
 }
 
+// Step wizard used in /shop, /event, appointment
+.o_wizard {
+    $o_wizard_circle_progress_size: 64px;
+
+    .o_wizard_step-active {
```
Comment (stefanorigano) [inline addons/website/static/src/scss/website.scss:1385]: Frontend convention: custom classes' words separator is always lowercase
 ```suggestion
    .o_wizard_step_active {
```

File: addons/website/static/src/scss/website.scss [hunk 14]
```diff
@@ -1378,6 +1378,36 @@ header {
...
+        @include media-breakpoint-up(md) {
+            color: $body-color;
+        }
+    }
+
+    .o_wizard_circle_progress {
+      width: $o_wizard_circle_progress_size;
+      height: $o_wizard_circle_progress_size;
+      line-height: $o_wizard_circle_progress_size;
+      box-shadow: inset 0px 0px 0px map-get($border-widths, 5) $gray-400;
```
Comment (stefanorigano) [inline addons/website/static/src/scss/website.scss:1395]: `$gray-400` works until the bg is grey 😉
I suggest:

```scss
.o_wizard_circle_progress {
      &, &:before {
          width: $o_wizard_circle_progress_size;
          height: $o_wizard_circle_progress_size;
          line-height: $o_wizard_circle_progress_size;
      }

      &:before {
          @include o-position-absolute(0, 0);
          box-shadow: inset 0 0 0 map-get($border-widths, 5) currentColor;
          opacity: 0.4; // Or "muted" opacity, I let you check
          content: "";
      }
```
Please note `currentColor` and `opacity`... also (minor), you can remove the units from "zero" values ;)

File: addons/website/static/src/scss/website.scss [hunk 15]
```diff
@@ -1378,6 +1378,36 @@ header {
...
+    }
+
+    .o_wizard_circle_progress-left .progress-bar {
+        border-radius: 0 ($o_wizard_circle_progress_size/2) ($o_wizard_circle_progress_size/2) 0;
```
Comment (stefanorigano) [inline addons/website/static/src/scss/website.scss:1399]: Convention: we normally multiply ;)

```suggestion
        border-radius: 0 ($o_wizard_circle_progress_size * .5) ($o_wizard_circle_progress_size * .5) 0;
```

File: addons/website/views/website_templates.xml [hunk 16]
```diff
@@ -2589,6 +2589,61 @@ Sitemap: <t t-esc="url_root"/>sitemap.xml
     </xpath>
 </template>
 
+<template id="website.step_wizard" name="Step Wizard">
+    <!-- REQUIRED WHEN CALLING -->
+    <!-- o_wizard_activestep = int active step (from 0) -->
```
Comment (stefanorigano) [inline addons/website/views/website_templates.xml:2594]: Can we set defaults rather than requiring values?

File: addons/website/views/website_templates.xml [hunk 17]
```diff
@@ -2589,6 +2589,61 @@ Sitemap: <t t-esc="url_root"/>sitemap.xml
...
+    <!-- o_wizard_steps_text = array of step text -->
+    <!-- o_wizard_steps_href = array of step href -->
+    <!-- o_wizard_back_default = <- back href if @first step of the stepper -->
+
+    <!-- OPTIONAL -->
+    <!-- o_wizard_back_custom = <- back href if @any step of the stepper -->
+
+    <t t-set="o_wizard_steps_in_deg" t-value="(o_wizard_activestep + 1) / len(o_wizard_steps_text) * 360"/>
```
Comment (stefanorigano) [inline addons/website/views/website_templates.xml:2602]: You can simplify contextual variable names to improve readability.
eg `o_wizard_activestep` ->`activestep` (or `_activestep`)

File: addons/website/views/website_templates.xml [hunk 18]
```diff
@@ -2589,6 +2589,61 @@ Sitemap: <t t-esc="url_root"/>sitemap.xml
...
+
+    <div class="o_wizard d-flex flex-wrap justify-content-end justify-content-sm-start my-3 my-sm-4">
+        <a class="align-self-center me-auto" t-attf-href="{{ o_wizard_back_custom if o_wizard_back_custom else o_wizard_steps_href[o_wizard_activestep - 1] if o_wizard_activestep &gt; 0 else o_wizard_back_default if o_wizard_back_default else '#'}}"><i class="fa fa-long-arrow-left me-2"/>Back</a>
```
Comment (stefanorigano) [inline addons/website/views/website_templates.xml:2605]: Could you move inner content in a new line?
It would improve code readability... also it would limit diffs if case we'll need to change just the content or the condition in the future
```suggestion
        <a class="align-self-center me-auto" t-attf-href="{{ o_wizard_back_custom if o_wizard_back_custom else o_wizard_steps_href[o_wizard_activestep - 1] if o_wizard_activestep &gt; 0 else o_wizard_back_default if o_wizard_back_default else '#'}}">
        <i class="fa fa-long-arrow-left me-2"/>Back</a>
```

File: addons/website/views/website_templates.xml [hunk 19]
```diff
@@ -2589,6 +2589,61 @@ Sitemap: <t t-esc="url_root"/>sitemap.xml
...
+        <div class="d-flex flex-column flex-md-row align-items-end align-items-md-start justify-content-center">
+            <t t-foreach="o_wizard_steps_text" t-as="step">
+                <a t-attf-class="d-none d-md-flex no-decoration {{'o_disabled' if step_index &gt; o_wizard_activestep else '' }}" t-att-href="o_wizard_steps_href[step_index]">
```
Comment (stefanorigano) [inline addons/website/views/website_templates.xml:2608]: You can
```suggestion
<a
    t-foreach="o_wizard_steps_text"
    t-as="step"
    t-attf-class="d-none d-md-flex no-decoration {{'o_disabled' if step_index &gt; o_wizard_activestep else '' }}
    t-att-href="o_wizard_steps_href[step_index]"
    >
```
Remember to add `title`attributes to links... in this case some `aria-role` as well I guess,

Comment (stefanorigano) [inline addons/website/views/website_templates.xml:2608]: no empty spaces, I let you check the rest 😉
```suggestion
                <a t-attf-class="d-none d-md-flex no-decoration {{'o_disabled' if step_index &gt; o_wizard_activestep else ''}}" t-att-href="o_wizard_steps_href[step_index]">
```

File: addons/website/views/website_templates.xml [hunk 20]
```diff
@@ -2589,6 +2589,61 @@ Sitemap: <t t-esc="url_root"/>sitemap.xml
...
+                    </div>
+                </a>
+            </t>
+            <div class="d-flex d-md-none flex-column align-items-end">
+                <div class="dropdown">
+                    <a t-attf-class="{{ 'dropdown-toggle' if o_wizard_activestep &gt; 0 else ''}} fw-bold" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+                        <t t-out="o_wizard_steps_text[o_wizard_activestep]"/>
+                    </a>
+                    <ul t-attf-class="dropdown-menu {{'d-none' if o_wizard_activestep == 0 else ''}}">
+                        <t t-foreach="o_wizard_steps_text" t-as="step">
```
Comment (stefanorigano) [inline addons/website/views/website_templates.xml:2623]: 😉
```suggestion
                        <li t-foreach="o_wizard_steps_text" t-as="step">
```

**Review by stefanorigano — CHANGES_REQUESTED** [review]

File: addons/website_sale/static/src/scss/website_sale.scss [hunk 21]
```diff
@@ -784,3 +764,45 @@ a.no-decoration {
...
 }
+
+.o_wsale_share_btn {
+    &::after {
+        display: none;
+    }
+}
+
+.o_wsale_share_menu {
+    min-width: 3rem;
```
Comment (stefanorigano) [inline addons/website_sale/static/src/scss/website_sale.scss:775]: We don't know the font-size... better to use `3em` for security 😉

File: addons/website_sale/static/src/scss/website_sale.scss [hunk 22]
```diff
@@ -784,3 +764,45 @@ a.no-decoration {
...
+}
+
+.o_total_border {
+    border: $border-width solid $border-color;
```
Comment (stefanorigano) [inline addons/website_sale/static/src/scss/website_sale.scss:779]: 🤔... 
```scss
    @include media-breakpoint-up(lg) {
        border: $border-width solid $border-color;
    }
 ```
 ?

File: addons/website_sale/static/src/scss/website_sale.scss [hunk 23]
```diff
@@ -784,3 +764,45 @@ a.no-decoration {
...
+    @include media-breakpoint-down(lg) {
+        border-width: 0;
+    }
+}
+
+.o_wsale_address_edit {
+    span[itemprop='name'] {
+        @extend h6;
```
Comment (stefanorigano) [inline addons/website_sale/static/src/scss/website_sale.scss:787]: Definitely avoid that 😬
Never `@extend` utility-classes, these operations take a huge amount on compilation time. Can you apply `h6`  class directly? If it's not possible just do:
```scss
span[itemprop='name'] {
    font-size: $h6-font-size;
    font-weight: $headings-font-weight;
    line-height: $headings-line-height;
}
```

File: addons/website_sale/views/templates.xml [hunk 24]
```diff
@@ -231,8 +231,8 @@
         <div t-attf-class="o_pricelist_dropdown dropdown #{_classes if hasPricelistDropdown else 'd-none'}">
             <t t-set="curr_pl" t-value="website.pricelist_id" />
 
-            <a role="button" href="#" t-attf-class="dropdown-toggle btn btn-{{navClass}}" data-bs-toggle="dropdown">
-                <t t-esc="curr_pl and curr_pl.name or ' - '" />
+            <a role="button" href="#" t-attf-class="dropdown-toggle d-flex align-items-center btn btn-{{navClass}}" data-bs-toggle="dropdown">
+                <span class="o_pricelist_dropdown_span d-inline-block text-truncate" t-esc="curr_pl and curr_pl.name or ' - '" />
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:235]: Let's double check if `text-truncate` is the reason for your JS tests failing 😉

File: addons/website_sale/views/templates.xml [hunk 25]
```diff
@@ -1381,102 +1369,90 @@
                 <strong>Warning!</strong> <t t-esc='website_sale_order._get_shop_warning()'/>
             </div>
         </t>
-        <table t-if="website_sale_order and website_sale_order.website_order_line" class="mb16 table table-striped table-sm js_cart_lines" id="cart_products">
+        <table t-if="website_sale_order and website_sale_order.website_order_line" class="js_cart_lines mb32 table table-borderless" id="cart_products">
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1372]: Classes order 😉
```suggestion
        <table t-if="website_sale_order and website_sale_order.website_order_line" class="js_cart_lines table table-borderless mb32" id="cart_products">
```

File: addons/website_sale/views/templates.xml [hunk 26]
```diff
@@ -1381,102 +1369,90 @@
...
-                                            <i class='fa fa-warning text-warning' t-att-title="line._get_shop_warning()" role="img" aria-label="Warning"/>
-                                            </a>
-                                        </t>
-                                        <a t-else='' t-attf-href="#" class="btn btn-link float_left js_add_cart_json d-none d-md-inline-block" aria-label="Add one" title="Add one">
-                                            <i class="fa fa-plus"></i>
+            <t t-foreach="website_sale_order.website_order_line" t-as="line">
+                <tr t-att-class="'optional_product info' if line.linked_line_id else None">
+                    <t t-if="not line.product_id" rowspan="2" class="td-img ps-0">
+                        <td rowspan="2" t-attf-class="td-img w-0 px-0 {{ 'border-bottom pb-4' if not line_last else '' }}"/>
+                        <td/>
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1378]: Let's double check if the second `<td/>` is necessary.
Eventually you may opt to use `colspan` in the first one instead 😉

File: addons/website_sale/views/templates.xml [hunk 27]
```diff
@@ -1381,102 +1369,90 @@
...
+                        <div class="css_quantity input-group mx-auto float-end">
+                            <t t-if="not line._is_not_sellable_line()">
+                                <t t-if="show_qty">
+                                    <a t-attf-href="#" class="btn btn-link js_add_cart_json d-inline-block border-end-0" aria-label="Remove one" title="Remove one">
+                                        <i class="fa fa-minus"></i>
+                                    </a>
+                                    <input type="text" class="js_quantity form-control quantity border-start-0 border-end-0" t-att-data-line-id="line.id" t-att-data-product-id="line.product_id.id" t-att-value="int(line.product_uom_qty) == line.product_uom_qty and int(line.product_uom_qty) or line.product_uom_qty" />
+                                    <t t-if="line._get_shop_warning(clear=False)">
+                                        <a t-attf-href="#" class="btn btn-link">
+                                        <i class='fa fa-warning text-warning' t-att-title="line._get_shop_warning()" role="img" aria-label="Warning"/>
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1411]: 😉
```suggestion
                                        <i class="fa fa-warning text-warning" t-att-title="line._get_shop_warning()" role="img" aria-label="Warning"/>
```

File: addons/website_sale/views/templates.xml [hunk 28]
```diff
@@ -1381,102 +1369,90 @@
...
                                         </a>
                                     </t>
-                                    <t t-else="">
-                                        <input type="hidden" class="js_quantity form-control quantity" t-att-data-line-id="line.id" t-att-data-product-id="line.product_id.id" t-att-value="int(line.product_uom_qty) == line.product_uom_qty and int(line.product_uom_qty) or line.product_uom_qty" />
-                                    </t>
+                                    <a t-else='' t-attf-href="#" class="btn btn-link float_left js_add_cart_json d-inline-block border-start-0" aria-label="Add one" title="Add one">
+                                        <i class="fa fa-plus fs-small"></i>
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1415]: I don't think that `fs-small` exists 😉
You probably needed `small` , remember to check the result on screen after applying
```suggestion
                                        <i class="fa fa-plus small"/>
```

File: addons/website_sale/views/templates.xml [hunk 29]
```diff
@@ -1381,102 +1369,90 @@
...
+                                    </a>
                                 </t>
                                 <t t-else="">
-                                    <span class="text-muted w-100" t-esc="int(line.product_uom_qty)"/>
-                                    <input type="hidden" class="js_quantity form-control quantity" t-att-data-line-id="line.id" t-att-data-product-id="line.product_id.id" t-att-value="line.product_uom_qty" />
+                                    <input type="hidden" class="js_quantity form-control quantity" t-att-data-line-id="line.id" t-att-data-product-id="line.product_id.id" t-att-value="int(line.product_uom_qty) == line.product_uom_qty and int(line.product_uom_qty) or line.product_uom_qty" />
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1419]: At this point I suggest a mass-search 😉
```suggestion
                                    <input type="hidden" class="js_quantity form-control quantity" t-att-data-line-id="line.id" t-att-data-product-id="line.product_id.id" t-att-value="int(line.product_uom_qty) == line.product_uom_qty and int(line.product_uom_qty) or line.product_uom_qty"/>
```

File: addons/website_sale/views/templates.xml [hunk 30]
```diff
@@ -1381,102 +1369,90 @@
...
+                            <t t-else="">
+                                <span class="text-muted w-100" t-esc="int(line.product_uom_qty)"/>
+                                <input type="hidden" class="js_quantity form-control quantity" t-att-data-line-id="line.id" t-att-data-product-id="line.product_id.id" t-att-value="line.product_uom_qty" />
+                            </t>
+                        </div>
+                    </td>
+                </tr>
+                <tr>
+                    <td t-attf-class="align-bottom pt-0 {{ 'border-bottom pb-4' if not line_last else '' }}">
+                        <a class="js_delete_product d-none d-md-inline-block" href='#' aria-label="Remove from cart" title="Remove from cart"><small>Remove</small></a>
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1431]: Let's save one HTML tag 😉
```suggestion
                        <a class="js_delete_product d-none d-md-inline-block small" href='#' aria-label="Remove from cart" title="Remove from cart">Remove</a>
```

File: addons/website_sale/views/templates.xml [hunk 31]
```diff
@@ -1381,102 +1369,90 @@
...
+                        <button type="button" role="button" class="js_delete_product btn btn-light d-inline-block d-md-none " title="Add to Wishlist" data-action="o_wishlist"><span class="fa fa-trash-o" role="img" aria-label="Add to wishlist"></span></button>
+                    </td>
+                    <td t-attf-class="td-price h6 fw-bold text-end pt-0 pe-0 align-bottom {{ 'border-bottom pb-4' if not line_last else '' }}" data-cell-type="price">
+                        <t t-set="combination" t-value="line.product_id.product_template_attribute_value_ids + line.product_no_variant_attribute_value_ids"/>
+                        <t t-set="combination_info" t-value="line.product_id.product_tmpl_id._get_combination_info(combination, pricelist=website_sale_order.pricelist_id, add_qty=line.product_uom_qty)"/>
+                        <span t-if="website.show_line_subtotals_tax_selection == 'tax_excluded'"
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1437]: Either you vertically align attributes between them or you use a fixed indentation.
I do not enforce one method over another, but please don't mix both :)

```xml
<span attribute-a="something"
      attribute-b="something"
      attribute-c="something"
      >

<span
    attribute-a="something"
    attribute-b="something"
    attribute-c="something"
    >

File: addons/website_sale/views/templates.xml [hunk 32]
```diff
@@ -1557,13 +1508,14 @@
 
     <!-- this template is the one when we mouse over "My Cart" on the top right -->
     <template id="cart_popover" name="Cart Popover">
+    <!-- let's not open the cart on hover if it's empty -->
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1511]: Indentation 😉
```suggestion
        <!-- let's not open the cart on hover if it's empty -->
```

File: addons/website_sale/views/templates.xml [hunk 33]
```diff
@@ -1572,61 +1524,81 @@
                             t-options="{'widget': 'image', 'qweb_img_responsive': True, 'class': 'rounded o_image_64_max mb-2'}"
                         />
                     </div>
-                    <div class="col-9">
-                        <div>
+                    <div class="col-9 ps-0">
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1527]: mmm....
Can't we use `g-0` on the main `row` and adjust paddings as needed?

Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:1527]: mmm....
Can't we use `g-0` on the main `row` and adjust paddings as needed?

**Review by stefanorigano — CHANGES_REQUESTED** [review]

File: addons/website_sale/views/templates.xml [hunk 34]
```diff
@@ -919,40 +909,58 @@
...
-                                <a href="#o_product_page_reviews" class="o_product_page_reviews_link text-decoration-none">
-                                    <t t-call="portal_rating.rating_widget_stars_static">
-                                        <t t-set="rating_avg" t-value="product.rating_avg"/>
-                                        <t t-set="trans_text_plural">%s reviews</t>
-                                        <t t-set="trans_text_singular">%s review</t>
-                                        <t t-set="rating_count" t-value="(trans_text_plural if product.rating_count > 1 else trans_text_singular) % product.rating_count"/>
+                        <div id="product_details" t-attf-class="{{ 'row m-0 p-0' if o_product_width_fw else 'offset-xl-1 col-lg-' + str(image_cols[1]) }} mt-lg-4">
+                            <div t-attf-class="d-flex">
+                                <div class="flex-grow-1 mb-3">
+                                    <h1 itemprop="name" t-field="product.name">Product Name</h1>
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:938]: -- not the exact line --

"100%" layout: I think that share and wishlist buttons should stay in the product's description column (left)

image

File: addons/website_sale/static/src/scss/product_configurator.scss [hunk 35]
```diff
@@ -65,11 +65,27 @@
         border-color: $input-border-color;
     }
 
+    .fa {
+        position: relative;
```
[the remark this answers is not in the document]
Comment (mano-odoo) [reply addons/website_sale/static/src/scss/product_configurator.scss:69]: Works fine for the position-relative and z-index but for the small the .fa class applies a font-size inherit which displays in 1rem because of the parent .btn class

File: addons/website_sale/static/src/scss/website_sale.scss (hunk 22 above)
Comment (mano-odoo) [reply addons/website_sale/static/src/scss/website_sale.scss]: This has to be done to have no border on the card on screen < than lg for the  `short_cart_summary` and apply the border on the accordion of the `cart_summary` using the same class

File: addons/website_sale/views/templates.xml [hunk 36]
```diff
@@ -2219,27 +2217,22 @@
...
-                  <tr id="empty">
-                      <t t-if='not no_rowspan'><td rowspan="10" class="border-0"/></t>
-                      <td class="col-md-2 col-3 border-0"></td>
-                      <td class="col-md-2 col-3 border-0" ></td>
-                  </tr>
                   <tr id="order_total_untaxed">
-                      <td class="text-end border-0">Subtotal:</td>
-                      <td class="text-xl-end border-0" >
+                      <td colspan="2" t-attf-class="border-0 ps-0 pt-0 {{'pb-3' if only_subtotal else 'pb-2'}} text-start text-muted">Subtotal</td>
+                      <td t-attf-class="text-end border-0 pe-0 pt-0 {{'pb-3' if only_subtotal else 'pb-2'}}" >
```
Comment (Feyensv) [inline addons/website_sale/views/templates.xml:2222]: what's this only_subtotal? it's not set anywhere :thinking:

File: addons/website_sale_product_configurator/views/website_sale_product_configurator_templates.xml [hunk 37]
```diff
@@ -1,10 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
-    <template id="configure_optional_products_website" inherit_id="website_sale_product_configurator.configure_optional_products">
-        <xpath expr="//th[hasclass('td-qty')]/span[hasclass('label')]" position="attributes">
-            <attribute name='t-if'>not request.is_frontend or (request.is_frontend and is_view_active('website_sale.product_quantity'))</attribute>
-        </xpath>
-    </template>
```
Comment (Feyensv) [inline addons/website_sale_product_configurator/views/website_sale_product_configurator_templates.xml:7]: ?

File: addons/website_sale/static/tests/tours/website_free_delivery.js [hunk 38]
```diff
@@ -34,14 +30,7 @@ registry.category("web_tour.tours").add('check_free_delivery', {
...
         },
-        {
-            content: "Select `Wire Transfer` payment method",
-            trigger: '#payment_method label:contains("Wire Transfer")',
-        },
-        {
-            content: "Click on Pay Now",
-            trigger: 'button[name="o_payment_submit_button"]:visible:not(:disabled)',
-        },
+        ...tourUtils.payWithDemo(),
```
Comment (Feyensv) [inline addons/website_sale/static/tests/tours/website_free_delivery.js:33]: no idea why it worked with transfer previously (since it doesn't confirm SO)

File: addons/website_sale/static/src/js/website_sale_tracking.js [hunk 39]
```diff
@@ -6,6 +6,7 @@ publicWidget.registry.websiteSaleTracking = publicWidget.Widget.extend({
     selector: '.oe_website_sale',
     events: {
         'click form[action="/shop/cart/update"] a.a-submit': '_onAddProductIntoCart',
+        // TODO ANVFE the two following events are probably broken
         'click a[href="/shop/checkout"]': '_onCheckoutStart',
         'click div.oe_cart a[href^="/web?redirect"][href$="/shop/checkout"]': '_onCustomerSignin',
```
Comment (Feyensv) [inline addons/website_sale/static/src/js/website_sale_tracking.js:11]: todo
